libqtile/backend/x11/core.py

```diff
@@ -236,6 +239,20 @@ def on_config_load(self, initial) -> None:
         """Assign windows to groups"""
         assert self.qtile is not None
 
+        # Check whether we need to subscribe for PropertyChange events
+        # (we use these to tell if the root pixmap has been updated which is
+        # required to update bars relying on pseudotransparency)
+        if self.qtile.config.x11_fake_transparency:
+            # Add PropertyChange to root window's event mask if it's not already there
+            if not self.eventmask & xcffib.xproto.EventMask.PropertyChange:
+                self.eventmask |= xcffib.xproto.EventMask.PropertyChange
+                self._root.set_attribute(eventmask=self.eventmask)
+        else:
+            # Remove PropertyChange from root window's event mask if it's there
+            if self.eventmask & xcffib.xproto.EventMask.PropertyChange:
+                self.eventmask &= ~xcffib.xproto.EventMask.PropertyChange
+                self._root.set_attribute(eventmask=self.eventmask)
```

Comment:
Is this what other projects use when they use pseudo transparency? Polybar perhaps?

Reply:
Yes - they listen for the same event. I didn't know that before I wrote the code but just checked!

Reply:
interestingly, polybar suggests that systray always uses pseudotransparency, even if the bar doesn't. I didn't think that was possible as you couldn't reparent a window with 24-bit depth into one with a 32-bit depth. I should look into that.

Reply:
Nice :) happy to have that in this PR or a follow up

Reply:
I might have a play around and see if I can add it here.

Reply:
Had a play. Will leave this for a follow-up!